### ISSUE TITLE
Navigate and manipulate ObjectLiteralExpressions

### ISSUE THREAD

**Issue by dsherret** [dsherret/ts-simple-ast#65]
Will add a description here later...

**Comment by gdeividas** [comment]
Hey @dsherret can you please add an example?

**Comment by dsherret** [comment]
Hey @gdeividas, currently this is not implemented. You can only navigate ObjectLiteralExpressions using basic commands currently. I will get to it sometime soon though.

**Comment by gregjacobs** [comment]
This would be great @dsherret. I'm looking to transform my current JS codebase which uses "classes" defined using [Class.js](https://github.com/gregjacobs/Class.js) into TypeScript classes, and it would be great to loop over all of an object literal's properties to extract the methods to populate a TS class.

**Comment by dsherret** [comment]
@gregjacobs lots of people have asked for this now. I'll up the priority and start working on it this week.

**Comment by gregjacobs** [comment]
@dsherret Awesome, thanks! 

Btw, thanks so much for this project! I just used it last night to set up a couple of transforms for some auto-generated interfaces created by Swagger-Codegen, and it was super easy and worked perfectly!

**Comment by gregjacobs** [comment]
I'll try out the `.getChildren()` btw in the meantime, but I've worked with the TypeScript AST before and let's just say: there's a good reason for `ts-simple-ast` :)

**Comment by dsherret** [comment]
@gregjacobs Nice, I'm glad it's been working well :)

haha! Yup, I know that all too well. I have pulled out many hairs trying to figure it out certain parts of the compiler.

**Comment by dsherret** [comment]
This is implemented for navigation in v0.96.0. I'll try to get manipulation in on the weekend or sometime next week.

```typescript
/// code.ts
const obj = {
    prop,
    prop2: 2,
    ...props
};

/// otherCode.ts (untested, typing from memory)
import Ast, {ObjectLiteralExpression, TypeGuards} from "ts-simple-ast";

const ast = ...;
const sourceFile = ast.getSourceFileOrThrow("code.ts");
const variableDeclaration = sourceFile.getVariableDeclarationOrThrow("obj");
// assertion won't be necessary after #138 is implemented
const ole = variableDeclaration.getInitializerIfKindOrThrow(ts.SyntaxKind.ObjectLiteralExpression) as ObjectLiteralExpression;

// loop over the properties
for (const prop of ole.getProperties()) {
    // prop can be any of the following:
    // PropertyAssignment | ShorthandPropertyAssignment | SpreadAssignment |
    // MethodDeclaration | GetAccessorDeclaration | SetAccessorDeclaration

    if (TypeGuards.isShorthandPropertyAssignment(prop)) {
        console.log(prop.getName()); // "prop"
    }
    else if (TypeGuards.isPropertyAssignment(prop)) {
        console.log(prop.getName()); // "prop2"
    }
    else if (TypeGuards.isSpreadAssignment(prop)) {
        console.log(prop.getExpression().getText()); // "props"
    }
}
```

**Comment by gregjacobs** [comment]
Awesomenessch! Will try this out! Thanks so much!

**Comment by dsherret** [comment]
This is implemented in 1.1.0. Documentation is [here](https://dsherret.github.io/ts-simple-ast/details/object-literal-expressions).

I have opened #148 for maybe making it easier to set an initializer as an object literal expression.
